fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/LocalFileBinary.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.modeshape;
+
+import static org.modeshape.jcr.api.JcrConstants.JCR_CONTENT;
+import static org.modeshape.jcr.api.JcrConstants.JCR_DATA;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.io.File;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.stream.Collectors;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import org.fcrepo.kernel.api.exception.InvalidChecksumException;
+import org.fcrepo.kernel.api.exception.RepositoryRuntimeException;
+import org.fcrepo.kernel.api.services.policy.StoragePolicyDecisionPoint;
+import org.fcrepo.kernel.modeshape.utils.FedoraTypesUtils;
+import org.slf4j.Logger;
+
+/**
+ * External binary from a local file
+ *
+ * @author bbpennel
+ * @since 11/29/2017
+ */
+public class LocalFileBinary extends UrlBinary {
+    private static final Logger LOGGER = getLogger(LocalFileBinary.class);
+
+    /**
+     * Constructs a LocalFileBinaryImpl
+     *
+     * @param node node
+     */
+    public LocalFileBinary(final Node node) {
+        super(node);
+    }
+
+    /*
+     * (non-Javadoc)
+      mlol* @see org.fcrepo.kernel.modeshape.FedoraBinaryImpl#setContent(java.io.InputStream, java.lang.String,
+     * java.util.Collection, java.lang.String, org.fcrepo.kernel.api.services.policy.StoragePolicyDecisionPoint)
+     */
+    @Override
+    public void setContent(final InputStream content, final String contentType,
```

Comment:
setContent should never be called on LocalFileBinary, only setExternalContent matters.  My suggestion would be to delete this and defer to `URLBinary`'s setContent method.